src/Modules/deRhamComplexes.jl

```diff
@@ -244,3 +244,57 @@ end
 function derivative(a::MPolyQuoRingElem, i::Int)
   return parent(a)(derivative(lift(a), i))
 end
+
+########################################################################
+# relative Kaehler differentials
+########################################################################
+
+function relative_kaehler_differentials(
+    R::MPolyAnyRing; 
+    base_variables::Vector{T}=elem_type(R)[]
+  ) where {T <: RingElem}
+  @assert all(x in gens(R) for x in base_variables) "given polynomials must be generators of the ring"
+  @assert _allunique(base_variables)
+  return _relative_kaehler_differentials(R, sort!([findfirst(==(y), gens(R)) for y in base_variables]))
+end
+
+function _relative_kaehler_differentials(
+    R::Union{MPolyRing, MPolyLocRing},
+    ind::Vector{Int}
+  )
+  n = ngens(R)
+  m = length(ind)
+  result = FreeMod(R, n-m)
+  ind_comp = [i for i in 1:n if !(i in ind)]
+  symb = symbols(R)[ind_comp]
+  result.S = [Symbol(:d, symb[i]) for i in 1:(n-m)]
```

Comment:
Messing with `result.S` after creating the module doesn't seem nice. Why not construct the module with the correct symbols from the start? It's even shorter.
```suggestion
  symb = [Symbol(:d, symb[i]) for i in 1:n if !(i in ind)]
  result = FreeMod(R, symb)
```

Reply:
Oh! I missed this feature. Thanks!